src/main/java/com/emc/ecs/tool/BucketWipe.java

```diff
@@ -176,154 +156,36 @@ public void run() {
         config.withIdentity(accessKey).withSecretKey(secretKey);
         S3Client client = new S3JerseyClient(config);
 
-        executor = new EnhancedThreadPoolExecutor(threads, new LinkedBlockingDeque<Runnable>(QUEUE_SIZE));
+        BucketWipeOperations bucketWipeOperations = new BucketWipeOperations(client, threads);
 
         if (sourceListFile != null) {
-            deleteAllObjectsWithList(client, bucket, sourceListFile);
+            bucketWipeOperations.deleteAllObjectsWithList(bucket, sourceListFile, wipeResult);
```

Comment:
these calls reference a static (`wipeResult`) inside a member method - maybe `wipeResult` should be a property of the object, retrieved via getter.  There should be a way to run multiple instances of BucketWipe in the same JVM and retrieve accurate status for each.